charts/nextcloud/templates/metrics/service.yaml

```diff
@@ -1,35 +1,38 @@
 {{- if .Values.metrics.enabled }}
+{{- with .Values.metrics.service }}
 ---
 apiVersion: v1
 kind: Service
 metadata:
-  name: {{ template "nextcloud.fullname" . }}-metrics
+  name: {{ template "nextcloud.fullname" $ }}-metrics
   labels:
-    app.kubernetes.io/name: {{ include "nextcloud.name" . }}
-    helm.sh/chart: {{ include "nextcloud.chart" . }}
-    app.kubernetes.io/instance: {{ .Release.Name }}
-    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/name: {{ include "nextcloud.name" $ }}
+    helm.sh/chart: {{ include "nextcloud.chart" $ }}
+    app.kubernetes.io/instance: {{ $.Release.Name }}
+    app.kubernetes.io/managed-by: {{ $.Release.Service }}
     app.kubernetes.io/component: metrics
-    {{- with .Values.metrics.service.labels }}
+    app.kubernetes.io/monitor: enabled
+    {{- with .labels }}
     {{- toYaml . | nindent 4 }}
     {{- end }}
-  {{- with .Values.metrics.service.annotations }}
+  {{- with .annotations }}
   annotations:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
-  type: {{ .Values.metrics.service.type }}
-  {{- if eq .Values.metrics.service.type "LoadBalancer" }}
-  {{- with .Values.metrics.service.loadBalancerIP }}
+  type: {{ .type }}
+  {{- if eq .type "LoadBalancer" }}
+  {{- with .loadBalancerIP }}
   loadBalancerIP: {{ . }}
   {{- end }}
   {{- end }}
+  selector:
+    app.kubernetes.io/name: {{ include "nextcloud.name" $ }}
+    app.kubernetes.io/instance: {{ $.Release.Name }}
+    app.kubernetes.io/component: metrics
   ports:
     - name: metrics
-      port: 9205
```

Comment:
Why is the port changed? This could be an unintended and unnecessary breaking change for some setups.

Reply:
Ok fine, just add it to the changelog to avoid surprises.